Makefile.common

```diff
@@ -133,6 +133,12 @@ CXXFLAGS += $(OLVL)
 CPPFLAGS += -D__TARGET_$(call uppercase,$(TARGET_FAMILY))
 CPPFLAGS += -D__CPU_$(call uppercase,$(TARGET_SUBFAMILY))
 
+WATCHDOG ?= 1
+ifneq ($(WATCHDOG), 0)
+  CPPFLAGS += -DWATCHDOG
+endif
+undefine WATCHDOG
```

Comment:
not sure about `undefine` as if it's passed by exported env variable - it would still be available in subprocesses (so a little bit misleading):

```make
WATCHDOG ?= 1
undefine WATCHDOG

$(info WATCHDOG = $(WATCHDOG))
$(info $(origin WATCHDOG))
$(info $(flavor WATCHDOG))
X=$(shell echo $$WATCHDOG)
$(info X = $(X))
```

for `WATCHDOG=3 make` returns:
```
WATCHDOG =
undefined
undefined
X = 3
```
(undefine works, env variable still present)

and for `make WATCHDOG=3` returns:
```
WATCHDOG = 3
command line
recursive
X =
```
(undefine doesn't work, env variable not present)

Maybe just don't undefine? :)

Reply:
Yep, wanted to be too clean. But nothing is that simple in make :)